tools/test_utils/src/MakeMeshes.cpp

```diff
@@ -53,11 +55,11 @@ ReadLegacyMeshFile(std::filesystem::path const& file_path)
     }
 
     std::size_t num_edges;
-    nc_inq_dim(ncidp, dimid, read_name.get(), &num_edges);
-    std::shared_ptr<double> node_x(new double[num_nodes]);
-    std::shared_ptr<double> node_y(new double[num_nodes]);
-    std::shared_ptr<int> edge_nodes(new int[num_edges * 2]);
-    std::shared_ptr<int> edge_type(new int[num_edges]);
+    nc_inq_dim(ncidp, dimid, read_name.data(), &num_edges);
+    std::shared_ptr<double> node_x(new double[num_nodes], std::default_delete<double[]>());
```

Comment:
Can't we simply use vectors instead of shared_ptr? this requires using the data method instead of get in the calls nc_* below.

Reply:
Yes, I did think about that, but went for the least disruptive option.
I will change these to std::vector.

Reply:
can we use make_shared, and avoid new?